### DISCUSSION TITLE
我想問戰場敵方名牌職業可以漢化嗎

### DISCUSSION THREAD

**Discussion by PongChan123** [GpointChen/FFXIVChnTextPatch-GP#254] [Q&A] [answered]
我想問戰場敵方名牌職業可以漢化嗎

**Comment by GpointChen** [comment]
有截圖嗎？方便我了解是哪裡需要漢化。

**Comment by PongChan123** [comment]
> 有截圖嗎？方便我了解是哪裡需要漢化。

其實就是戰場入面敵人頭上的職業名 全都是英文

**Comment by PongChan123** [comment]
我見國服是中文
你漢化包應該有國服中文
理論上戰場入面敵人名稱都跟國服一樣
難到我漢化步驟有問題

**Comment by PongChan123** [comment]
![5D0935C2-0B73-4E3C-B554-76A783A8826F](https://user-images.githubusercontent.com/99170448/209306315-e8792ad0-3e5c-4820-8a84-306622430a92.jpeg)
例如：紅色字/頭頂血條的紅色字

**Comment by chun1324** [comment]
你意思是將72人PVP裡其他角色的「名稱」<<職業名>> 翻譯成中文? 例如: <<戰士>> <<學者>>
角色名稱搞不好是從伺服器端派發, 不清楚能不能改

**Comment by PongChan123** [comment]
> 你意思是將72人PVP裡其他角色的「名稱」<<職業名>> 翻譯成中文? 例如: <<戰士>> <<學者>>
> 角色名稱搞不好是從伺服器端派發, 不清楚能不能改

對呀
希望能改

**Comment by GpointChen** [comment, marked as answer]
剛才研究了一下，應該是可以改的，參考下面這個日文的Addon.csv：

```
3490,《<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>》
3491,《<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>》
3492,«<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>»
```


將以上這三行換成國服匯出的Addon.csv這三行：

```
3490,"[<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
3491,"[<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
3492," [<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
```

應該就可以了。須注意，更改時可能不會只改到PVP職業名顯示，其他與職業名顯示相關的地方應該也會變。如果你有需要，可以自行更改，然後還原、重新漢化。


## 以下是解釋。

上面的東西用Godbert軟體翻譯一下就是（以3490那行為例）：

日文的：
* 《<If(IntegerParameter(3))><Sheet(ClassJob,IntegerParameter(1),1)/><Else/><Sheet(ClassJob,IntegerParameter(1),27)/></If>》

當某個值為3的時候，顯示ClassJob表中key=1那個column的值，反之則顯示key=27那個column的值。

這是ClassJob.csv的一部份。

```csv
key,0,1,2,27
#,Name,Abbreviation,,Name{English}
offset,0,4,8,16
int32,str,str,str,str
0,冒險者,ADV,,Adventurer
1,劍術師,GLA,劍,Gladiator
2,格鬥家,PGL,格,Pugilist
3,斧術師,MRD,斧,Marauder
4,槍術師,LNC,槍,Lancer
5,弓箭手,ARC,弓,Archer
6,幻術師,CNJ,幻,Conjurer
7,咒術師,THM,咒,Thaumaturge
8,刻木匠,CRP,木,Carpenter
9,鍛鐵匠,BSM,鍛,Blacksmith
10,鑄甲匠,ARM,甲,Armorer
```

也就是說，當某個值為3的時候，顯示ADV、GLA這種英文簡稱；反之則顯示Adventurer、Gladiator這種英文職業名。

同樣看看中文的Addon.csv：
* [<If(IntegerParameter(3))><Sheet(ClassJob,IntegerParameter(1),2)/><Else/><Sheet(ClassJob,IntegerParameter(1),0)/></If>]

這句話的意思是當某個值為3的時候，顯示key=2的直行的劍、格這種簡稱，反之則顯示key=0的直行的劍術師、格鬥家這種翻譯後的職業名。所以國際服和中國服顯示的才會不一樣。

**Comment by PongChan123** [comment]
> 剛才研究了一下，應該是可以改的，參考下面這個日文的Addon.csv：
> 
> ```
> 3490,《<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>》
> 3491,《<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>》
> 3492,«<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020203><hex:FF12><hex:02280EFF09>ClassJob<hex:E8021C03><hex:03>»
> ```
> 
> 將以上這三行換成國服匯出的Addon.csv這三行：
> 
> ```
> 3490,"[<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
> 3491,"[<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
> 3492," [<hex:020829E804FF12><hex:02280EFF09>ClassJob<hex:E8020303><hex:FF12><hex:02280EFF09>ClassJob<hex:E8020103><hex:03>]"
> ```
> 
> 應該就可以了。須注意，更改時可能不會只改到PVP職業名顯示，其他與職業名顯示相關的地方應該也會變。如果你有需要，可以自行更改，然後還原、重新漢化。
> 
> ## 以下是解釋。
> 上面的東西用Godbert軟體翻譯一下就是（以3490那行為例）：
> 
> 日文的：
> 
> * 《<If(IntegerParameter(3))><Sheet(ClassJob,IntegerParameter(1),1)/><Sheet(ClassJob,IntegerParameter(1),27)/>》
> 
> 當某個值為3的時候，顯示ClassJob表中key=1那個column的值，反之則顯示key=27那個column的值。
> 
> 這是ClassJob.csv的一部份。
> 
> ```
> key,0,1,2,27
> #,Name,Abbreviation,,Name{English}
> offset,0,4,8,16
> int32,str,str,str,str
> 0,冒險者,ADV,,Adventurer
> 1,劍術師,GLA,劍,Gladiator
> 2,格鬥家,PGL,格,Pugilist
> 3,斧術師,MRD,斧,Marauder
> 4,槍術師,LNC,槍,Lancer
> 5,弓箭手,ARC,弓,Archer
> 6,幻術師,CNJ,幻,Conjurer
> 7,咒術師,THM,咒,Thaumaturge
> 8,刻木匠,CRP,木,Carpenter
> 9,鍛鐵匠,BSM,鍛,Blacksmith
> 10,鑄甲匠,ARM,甲,Armorer
> ```
> 
> 也就是說，當某個值為3的時候，顯示ADV、GLA這種英文簡稱；反之則顯示Adventurer、Gladiator這種英文職業名。
> 
> 同樣看看中文的Addon.csv：
> 
> * [<If(IntegerParameter(3))><Sheet(ClassJob,IntegerParameter(1),2)/><Sheet(ClassJob,IntegerParameter(1),0)/>]
> 
> 這句話的意思是當某個值為3的時候，顯示key=2的直行的劍、格這種簡稱，反之則顯示key=0的直行的劍術師、格鬥家這種翻譯後的職業名。所以國際服和中國服顯示的才會不一樣。

最好加入以後的漢化🙏

**Reply by GpointChen** [reply]
之後如果確認不會影響到其他地方的顯示或產生bug的話應該會加。

**Reply by PongChan123** [reply]
我自己用了一個晚上 無BUG

**Reply by Manbbb** [reply]
這部份可以自行選擇嗎？一堆中文反而更暈

**Comment by PongChan123** [comment]
我自己用了一個晚上 無BUG
